code/datums/helper_datums/getrev.dm

```diff
@@ -21,13 +23,63 @@ var/global/datum/getrev/revdata = new()
 	if(testmerge.len)
 		world.log << commit
 		for(var/line in testmerge)
-			if(line)
-				world.log << "Test merge active of PR #[line]"
+			world.log << "Test merge active of PR #[line]"
 		world.log << "Based off master commit [parentcommit]"
 	else
 		world.log << parentcommit
 	world.log << "Current map - [MAP_NAME]" //can't think of anywhere better to put it
 
+/datum/getrev/proc/DownloadPRDetails()
+	if(!text2num(config.githubrepoid))
+		world.log << "Invalid github repo id: [config.githubrepoid]"
+		return
+	for(var/line in testmerge)
+		//"This has got to be the ugliest hack I have ever done"
+		//warning, here be dragons
+		/*
+			                |  @___oo
+			      /\  /\   / (__,,,,|
+			     ) /^\) ^\/ _)
+			     )   /^\/   _)
+			     )   _ /  / _)
+			 /\  )/\/ ||  | )_)
+			<  >      |(,,) )__)
+			 ||      /    \)___)\
+			 | \____(      )___) )___
+			  \______(_______;;; __;;;
+			*/
+		if(!text2num(line))
+			world.log << "Invalid PR number: [line]"
+			return
+		var/url = "https://api.github.com/repositories/[config.githubrepoid]/pulls/[line].json"
+		var/temp_file = "prcheck[line].json"
+		var/command = "powershell -Command \"curl -O [temp_file] [url]\""
+		world.log << "Running command: [command]"
+		if(shell(command) != 0)
+			return
+
+		var/f = file(temp_file)
+		if(!f)
+			return
+		testmerge[line] = json_decode(file2text(f))
+		f = null
+		fdel(temp_file)
+
+		if(!testmerge[line])
+			return
+		CHECK_TICK
+	has_pr_details = TRUE
+
+/datum/getrev/proc/GetTestMergeInfo()
+	if(!testmerge.len)
+		return ""
+	. = "The following pull requests are currently test merged:<br/>"
+	for(var/line in testmerge)
+		var/details = ""
+		if(has_pr_details)
+			details = ": '" + testmerge[line]["title"] + "' by " + testmerge[line]["user"]["login"]
+		. += "<a href='[config.githuburl]/pull/[line]'>#[line][details]</a><br/>"

```

Comment:
Needs html sanitation or somebody could inject html into the chatbox by putting it in the title.